OPTIONS on AI endpoints returns 500 from Jersey WADL JAXB serialization failure

## Repro

```bash
curl -X OPTIONS http://localhost:8080/rest/saiku/api/ai/cubes
```

## Expected
HTTP 200 with `Allow: GET, POST, ...` header — standard JAX-RS OPTIONS response — or 405 Method Not Allowed.

## Actual
HTTP 500, empty body. Log shows Jersey's auto-WADL response failing JAXB marshalling:

```
Caused by: jakarta.xml.bind.JAXBException: class com.sun.research.ws.wadl.Application nor any of its super class is known to this context.
    at org.glassfish.jaxb.runtime.v2.runtime.JAXBContextImpl.getBeanInfo
    ...
```

## Why it matters

- CORS preflight requests use OPTIONS. If saiku is ever called from a browser cross-origin, every preflight will 500. Currently moot because saiku gates everything behind form login, but worth noting for future CORS work.
- The 500 leaks no information but breaks any client that uses OPTIONS for capability discovery.

## Probable cause

Jersey's auto-OPTIONS handler returns a WADL document (XML). The WADL marshalling depends on the `org.glassfish.jersey.media.jersey-wadl-doclet` / `jakarta.xml.bind.JAXBContext` setup. Either:
- A missing dependency (`com.sun.research.ws.wadl` types not in the JAXBContext)
- A Jakarta EE 10 migration gap (old `javax.xml.bind` mappings not ported)

## Suggested fix

Either register a custom `@Provider OPTIONS` handler in `AiQueryResource` and the other Jersey resources that returns `Allow` headers explicitly, or disable Jersey's WADL feature with `ResourceConfig.property("jersey.config.server.wadl.disableWadl", true)` in `SaikuJerseyApplication`.

## Environment
- saiku feat/ai-query-api @ b3725291
- Jersey 3.1.x via Jetty 12
- Discovered via ralph-loop fuzz iter 248, 2026-05-15

--- a/saiku-core/saiku-web/src/main/java/org/saiku/web/rest/SaikuJerseyApplication.java
+++ b/saiku-core/saiku-web/src/main/java/org/saiku/web/rest/SaikuJerseyApplication.java
@@ -11,6 +11,14 @@ import org.springframework.web.context.support.WebApplicationContextUtils;
 public class SaikuJerseyApplication extends ResourceConfig {
 
     public SaikuJerseyApplication(@Context ServletContext servletContext) {
+        // saiku#806: disable Jersey's auto-WADL OPTIONS response. The auto
+        // handler tries to marshal a com.sun.research.ws.wadl.Application
+        // document via JAXB, but those classes aren't on the JAXBContext
+        // we ship under Jakarta EE 10 — every OPTIONS request 500s out of
+        // the auto-WADL marshaller. With this off, Jersey falls back to a
+        // plain Allow-header response for OPTIONS, which is what every
+        // standard JAX-RS client (and CORS preflight) expects.
+        property("jersey.config.server.wadl.disableWadl", true);
         register(MultiPartFeature.class);
 
         ApplicationContext ctx = WebApplicationContextUtils.getRequiredWebApplicationContext(servletContext);

